consensus/istanbul/backend/api.go

```diff
@@ -248,17 +280,12 @@ func (api *APIExtension) getConsensusInfo(block *types.Block) (ConsensusInfo, er
 
 	// get origin proposer at 0 round.
 	originProposer := common.Address{}
-	lastProposer := common.Address{}
-	// TODO-Klaytn add the logic to get the last proposer for other policies. Weighted Random doesn't need it.
-	if istanbul.ProposerPolicy(snap.Policy) == istanbul.WeightedRandom {
-		newValSet := snap.ValSet.Copy()
-		newValSet.CalcProposer(lastProposer, 0)
-		originProposer = newValSet.GetProposer().Address()
-	} else {
-		logger.Warn("getConsensusInfo does not support to get origin proposal on", "proposerPolicy", snap.Policy)
-	}
+	_, lastProposer := api.istanbul.LastProposal()
```

Comment:
Isn't `lastProposer` identical to `proposer`?

Reply:
This function returns the parent block proposer.
I missed to consider the given block number. I will fix it. Thanks.